aodndata/gsla/handler.py

```diff
@@ -148,13 +163,20 @@ def get_previous_version_creation_date(self, filepath):
 
     @staticmethod
     def dest_path(filepath):
+        " Netcdf only as an input. Not nc.gz"
         file_basename = os.path.basename(filepath)
         gsla_type = get_gsla_type(filepath)
 
         if GSLA_REGEX_YEARLY.match(file_basename):
-            return os.path.join(GSLA_PREFIX_PATH, gsla_type, file_basename)
+            dest_path_val = os.path.join(GSLA_PREFIX_PATH, gsla_type, file_basename)
 
         else:
             fields = get_pattern_subgroups_from_string(file_basename, GSLA_REGEX)
             gsla_year = datetime.strptime(fields['nc_time_cov_start'], '%Y%m%dT%H%M%SZ').year
-            return os.path.join(GSLA_PREFIX_PATH, gsla_type, str(gsla_year), file_basename)
+            dest_path_val = os.path.join(GSLA_PREFIX_PATH, gsla_type, str(gsla_year), file_basename)
+
+        # destination path should always be for nc.gz files. So we force it
+        if dest_path_val.endswith('.nc'):
```

Comment:
If the if/else only controls a single value, you can just use a ternary style statement:

```python
return '{val}.gz'.format(val=dest_path_val) if dest_path_val.endswith('.nc') else dest_path_val
```